spring-cloud-stream/src/test/java/org/springframework/cloud/stream/binder/BinderFactoryConfigurationTests.java

```diff
@@ -148,6 +148,38 @@ public void loadBinderTypeRegistryWithTwoBinders() throws Exception {
 	}
 
 	@Test
+	public void loadBinderTypeRegistryWithUserDefinedBinder() throws Exception {
+
+		ConfigurableApplicationContext context = createBinderTestContext(
+				new String[] { "binder1"},
+				"spring.cloud.stream.binders.custom.type=binder1",
+				"spring.cloud.stream.binders.custom.environment.binder1.name=foo",
+				"spring.cloud.stream.binders.custom.applicationProvided=false",
```

Comment:
Is this supposed to use `true` as this test case suggests `user-defined` binder? or may be we just need another test case to test `user-defined` binder to be chosen as the `defaultBinder`?